Index: app/src/main/java/com/yx/srtool/Activity/RocketDesignActivity.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- app/src/main/java/com/yx/srtool/Activity/RocketDesignActivity.java	(revision 47e437015c243ef1db46a1a21692136a088524ec)
+++ app/src/main/java/com/yx/srtool/Activity/RocketDesignActivity.java	(revision )
@@ -1,6 +1,8 @@
 package com.yx.srtool.Activity;
 
 import android.animation.ValueAnimator;
+import android.content.Intent;
+import android.net.Uri;
 import android.os.Bundle;
 import android.support.design.widget.TabLayout;
 import android.support.v4.view.PagerAdapter;
@@ -231,6 +233,13 @@
                 //保存
                 FileUtil.write("ships/"+ MainActivity.ship_name,RocketView.ship);
                 Toast.makeText(this, "保存成功", Toast.LENGTH_SHORT).show();
+                return true;
+            case R.id.menu_p:
+                //导出图片
+                RocketView.saveimage(this);
+
+                Toast.makeText(this, "图片已保存至sd卡根目录", Toast.LENGTH_SHORT).show();
+
                 return true;
         }
 
Index: app/src/main/AndroidManifest.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- app/src/main/AndroidManifest.xml	(revision 47e437015c243ef1db46a1a21692136a088524ec)
+++ app/src/main/AndroidManifest.xml	(revision )
@@ -33,6 +33,8 @@
                 <category android:name="android.intent.category.DEFAULT" />
                 <category android:name="android.intent.category.BROWSABLE" />
                 <data android:scheme="srtool"/>
+                <data android:scheme="http" />
+                <data android:scheme="https" />
                 <data android:host="jundroo.com" />
                 <data android:pathPrefix="/ViewShip.html"/>
             </intent-filter>
@@ -43,7 +45,7 @@
             android:icon="@drawable/icon_sr"
             android:launchMode="singleTop"
             android:screenOrientation="unspecified"
-            android:theme="@style/Theme.Dialog"
+            android:theme="@style/Dialog"
             android:configChanges="keyboardHidden|orientation|screenSize">
             <intent-filter>
                 <action android:name="android.intent.action.VIEW"/>
@@ -62,6 +64,8 @@
                 <!--注意下面是 content -->
                 <data android:scheme="file" />
                 <data android:scheme="content" />
+                <data android:scheme="http" />
+                <data android:scheme="https" />
                 <data android:host="*" />
                 <data android:mimeType="*/*" />
                 <data android:pathPattern=".*\\.srmod" />
@@ -72,8 +76,32 @@
                 <category android:name="android.intent.category.BROWSABLE"/>
                 <data android:scheme="file" />
                 <data android:scheme="content" />
+                <data android:scheme="http" />
+                <data android:scheme="https" />
                 <data android:host="*" />
                 <data android:mimeType="	text/*" />
+            </intent-filter>
+        </activity>
+        <activity
+            android:name=".i.ShipPoint"
+            android:label="简单火箭存档"
+            android:icon="@drawable/icon_sr"
+            android:launchMode="singleTop"
+            android:screenOrientation="unspecified"
+            android:theme="@style/Dialog"
+            android:configChanges="keyboardHidden|orientation|screenSize">
+            <intent-filter>
+                <action android:name="android.intent.action.VIEW"/>
+                <category android:name="android.intent.category.DEFAULT"/>
+                <category android:name="android.intent.category.BROWSABLE"/>
+                <data android:scheme="file" />
+                <data android:scheme="content" />
+                <data android:scheme="http" />
+                <data android:scheme="https" />
+                <data android:host="*" />
+                <data android:mimeType="	text/*" />
+                <data android:mimeType="application/hcy" />
+                <data android:mimeType="application/octet-stream" />
             </intent-filter>
         </activity>
         <activity android:name=".i.AboutPoint"></activity>
@@ -89,6 +117,7 @@
             android:hardwareAccelerated="false"
             android:largeHeap="true">
         </activity>
+
         <provider
             android:name="android.support.v4.content.FileProvider"
             android:authorities="com.yx.srtool.fileprovider"
Index: app/build.gradle
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- app/build.gradle	(revision 47e437015c243ef1db46a1a21692136a088524ec)
+++ app/build.gradle	(revision )
@@ -9,8 +9,8 @@
         applicationId "com.yx.srtool"
         minSdkVersion 16
         targetSdkVersion 23
-        versionCode 1
-        versionName "beta1"
+        versionCode 2
+        versionName "beta2"
         testInstrumentationRunner "android.support.test.runner.AndroidJUnitRunner"
     }
     /*
Index: app/src/main/res/values/colors.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- app/src/main/res/values/colors.xml	(revision 47e437015c243ef1db46a1a21692136a088524ec)
+++ app/src/main/res/values/colors.xml	(revision )
@@ -3,5 +3,5 @@
     <color name="colorPrimary">#303030</color>
     <color name="colorPrimaryDark">#252525</color>
     <color name="colorAccent">#1E88E5</color>
-
+    <color name="touming">#00ffffff</color>
 </resources>
Index: app/src/main/res/menu/menu_rocket.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- app/src/main/res/menu/menu_rocket.xml	(revision 47e437015c243ef1db46a1a21692136a088524ec)
+++ app/src/main/res/menu/menu_rocket.xml	(revision )
@@ -1,6 +1,11 @@
 <menu xmlns:android="http://schemas.android.com/apk/res/android"
     xmlns:app="http://schemas.android.com/apk/res-auto">
     <item
+        android:id="@+id/menu_p"
+        android:orderInCategory="100"
+        android:title="输出图片"
+        app:showAsAction="never"/>
+    <item
         android:id="@+id/menu_baochun"
         android:orderInCategory="100"
         android:title="保存"
Index: app/src/main/res/values/styles.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- app/src/main/res/values/styles.xml	(revision 47e437015c243ef1db46a1a21692136a088524ec)
+++ app/src/main/res/values/styles.xml	(revision )
@@ -23,12 +23,11 @@
         <item name="colorButtonNormal">@color/colorPrimary</item>
     </style>
 
-    <style name="Theme.Dialog" parent="android:style/Theme.Dialog">
+    <style name="Dialog" parent="Theme.AppCompat.Light.NoActionBar">
+        <item name="android:windowBackground">@color/touming</item>
         <item name="android:windowNoTitle">true</item>
-        <item name="android:background">#00000000</item>
-        <item name="android:windowBackground">@android:color/transparent</item>
-        <item name="android:colorBackgroundCacheHint">@null</item>
         <item name="android:windowIsTranslucent">true</item>
+        <item name="android:windowAnimationStyle">@android:style/Animation.Translucent</item>
     </style>
 
     <style name="noTitleDialog" parent="Theme.AppCompat.Light.NoActionBar">
Index: app/src/main/java/com/yx/srtool/Activity/MainActivity.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- app/src/main/java/com/yx/srtool/Activity/MainActivity.java	(revision 47e437015c243ef1db46a1a21692136a088524ec)
+++ app/src/main/java/com/yx/srtool/Activity/MainActivity.java	(revision )
@@ -102,7 +102,6 @@
             init_permission();
             //初始化
             init();
-
         }catch (Exception e){
         }
     }
@@ -155,6 +154,14 @@
                 DownloadFile(ID);
             }
         }
+        if(intent.getStringExtra("mod").indexOf("true")!=-1){
+            Toast.makeText(TContext, "导入Mod", Toast.LENGTH_SHORT).show();
+            intent.setAction(Intent.ACTION_VIEW);
+            intent.setClassName("com.jundroo.simplerockets", "com.jundroo.simplerockets.MainActivity");
+            Uri uri = Uri.fromFile(new File(intent.getStringExtra("path")));
+            intent.setData(uri);
+            startActivityForResult(intent, 1);
+        }
     }
 
     /**
@@ -253,7 +260,7 @@
         String data = null;
         try {
             //读取配置文件
-            data = FileUtil.read("SimpleRockets-UserSettings.txt");
+            data = FileUtil.read(MainActivity.path+"SimpleRockets-UserSettings.txt");
             String data2 = Util.substring(data,"fpsEnabled=\"","\"");
             if (data2!=null && data2.equals("1")){
                 CheckBox_FPS.setChecked(true);
@@ -336,16 +343,17 @@
         //赋值WindowManager&LayoutParam.
         params = new WindowManager.LayoutParams();
         windowManager = (WindowManager) getApplication().getSystemService(Context.WINDOW_SERVICE);
+        /*
         //设置type.系统提示型窗口，一般都在应用程序窗口之上.
-/*                if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {//https://blog.csdn.net/qq_23374873/article/details/80718948
-                    //关于 8.0以上
-                    params.type = WindowManager.LayoutParams.TYPE_APPLICATION_OVERLAY;
-                }else {
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {//https://blog.csdn.net/qq_23374873/article/details/80718948
+            //关于 8.0以上
+            params.type = WindowManager.LayoutParams.TYPE_APPLICATION_OVERLAY;
+        }else {
 
-                }*/
+        }
+        */
         //关于 8.0以下
         params.type = WindowManager.LayoutParams.TYPE_SYSTEM_ALERT;
-
         //设置效果为背景透明.
         params.format = PixelFormat.RGBA_8888;
         //设置flags.不可聚焦及不可使用按钮对悬浮窗进行操控.
@@ -458,7 +466,7 @@
                         windowManager.updateViewLayout(toucherLayout, params);
                         handler_time.postDelayed(runnable,5);
                     }else {
-                        Toast.makeText(MainActivity.this, "透明", Toast.LENGTH_SHORT).show();
+                        Toast.makeText(MainActivity.this, "滑动", Toast.LENGTH_SHORT).show();
                     }
                 }else {
                     if(params.x>0){
@@ -466,7 +474,7 @@
                         windowManager.updateViewLayout(toucherLayout, params);
                         handler_time.postDelayed(runnable,5);
                     }else {
-                        Toast.makeText(MainActivity.this, "透明", Toast.LENGTH_SHORT).show();
+                        Toast.makeText(MainActivity.this, "滑动", Toast.LENGTH_SHORT).show();
                     }
                 }
                 //开始透明
@@ -481,7 +489,7 @@
         //清除载具目录数据
         data_modifier.clear();
         try {
-            String data = FileUtil.read("Sandbox.xml");
+            String data = FileUtil.read(MainActivity.path+"Sandbox.xml");
             //分割字串符
             String[] ShipNode = data.split("</ShipNode>");
             Log.e("Sandbox",data);
Index: app/src/main/res/layout/view_gridview.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- app/src/main/res/layout/view_gridview.xml	(revision 47e437015c243ef1db46a1a21692136a088524ec)
+++ app/src/main/res/layout/view_gridview.xml	(revision )
@@ -12,5 +12,7 @@
         android:layout_width="match_parent"
         android:layout_height="wrap_content"
         android:id="@+id/Button_Modifiers"
+        android:enabled="false"
+
         android:text="坐标修改器"/>
 </LinearLayout>
\ No newline at end of file
Index: app/proguard-rules.pro
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- app/proguard-rules.pro	(revision 47e437015c243ef1db46a1a21692136a088524ec)
+++ app/proguard-rules.pro	(revision )
@@ -6,6 +6,7 @@
 #混淆时采用的算法
 -optimizations !code/simplification/arithmetic,!field/*,!class/merging/*
 
+-ignorewarnings
 
 
 # 修改包名
Index: app/src/main/java/com/yx/srtool/Activity/ModActivity.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- app/src/main/java/com/yx/srtool/Activity/ModActivity.java	(revision 47e437015c243ef1db46a1a21692136a088524ec)
+++ app/src/main/java/com/yx/srtool/Activity/ModActivity.java	(revision )
@@ -11,7 +11,6 @@
 import android.os.Message;
 import android.provider.Settings;
 import android.support.v7.app.AppCompatActivity;
-import android.util.Log;
 import android.view.LayoutInflater;
 import android.view.View;
 import android.view.WindowManager;
@@ -25,13 +24,14 @@
 import com.yx.srtool.i.MainPoint;
 
 import java.io.BufferedReader;
-import java.io.File;
 import java.io.InputStreamReader;
 
 import ru.noties.markwon.Markwon;
 import ru.noties.markwon.html.HtmlPlugin;
 import ru.noties.markwon.image.ImagesPlugin;
 
+import static com.yx.srtool.Activity.MainActivity.getContext;
+
 /**
  * Created by Yx on 2019/4/25.
  */
@@ -39,10 +39,6 @@
 public class ModActivity extends AppCompatActivity {
     //String path;
     private ProgressDialog Dialog;
-
-    public ModActivity() {
-    }
-
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
@@ -154,7 +150,15 @@
             openSR.setOnClickListener(new View.OnClickListener() {
                 @Override
                 public void onClick(View v) {
-                /*如果API是24以上，打开注释
+                    dialog.cancel();//关闭
+                    Intent intent_activity =
+                            new Intent(ModActivity.this,
+                            MainPoint.class);
+                    intent_activity.putExtra("mod","true");
+                    intent_activity.putExtra("path",path);
+                    startActivity(intent_activity);
+                    /*
+                //如果API是24以上，打开注释
                 Intent intent = new Intent();
                 intent.setAction(Intent.ACTION_VIEW);
                 intent.setClassName("com.jundroo.simplerockets", "com.jundroo.simplerockets.MainActivity");
@@ -170,7 +174,9 @@
                 }
                 Log.e("uri",uri+"");
                 startActivityForResult(intent, 1);
+                //startActivity(intent);
                 */
+                    /*
                     Intent intent = new Intent();
                     intent.setAction(Intent.ACTION_VIEW);
                     intent.setClassName("com.jundroo.simplerockets", "com.jundroo.simplerockets.MainActivity");
@@ -180,9 +186,11 @@
                     startActivityForResult(intent, 1);
                     dialog.cancel();//关闭
                     //startActivity(intent);
+                    finish();
+                    */
                 }
             });
-        /*
+            /*
         if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {//https://blog.csdn.net/qq_23374873/article/details/80718948
             //关于 8.0以上
             dialog.getWindow().setType(
@@ -193,11 +201,11 @@
                     (WindowManager.LayoutParams.TYPE_SYSTEM_ALERT));
         }
         */
+        //关于 8.0以下
             dialog.getWindow().setType(
                     (WindowManager.LayoutParams.TYPE_SYSTEM_ALERT));
-
-            Handler handler = new Handler(Looper.getMainLooper());
-            handler.post(new Runnable() {
+        Handler handler = new Handler(Looper.getMainLooper());
+        handler.post(new Runnable() {
                 public void run() {
                     dialog.show();
                 }
Index: app/src/main/java/com/yx/srtool/i/MainPoint.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- app/src/main/java/com/yx/srtool/i/MainPoint.java	(revision 47e437015c243ef1db46a1a21692136a088524ec)
+++ app/src/main/java/com/yx/srtool/i/MainPoint.java	(revision )
@@ -1,9 +1,13 @@
 package com.yx.srtool.i;
 
+import android.content.Intent;
+import android.net.Uri;
 import android.os.Bundle;
 
 import com.yx.srtool.Activity.MainActivity;
 
+import java.io.File;
+
 /**
  * Created by Yx on 2019/5/9.
  */
@@ -33,6 +37,7 @@
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
+
         new over();
     }
 }
Index: app/src/main/java/com/yx/srtool/View/RocketView.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- app/src/main/java/com/yx/srtool/View/RocketView.java	(revision 47e437015c243ef1db46a1a21692136a088524ec)
+++ app/src/main/java/com/yx/srtool/View/RocketView.java	(revision )
@@ -1,24 +1,33 @@
 package com.yx.srtool.View;
 
 import android.content.Context;
+import android.content.Intent;
 import android.graphics.Bitmap;
 import android.graphics.Canvas;
 import android.graphics.Color;
 import android.graphics.Matrix;
 import android.graphics.Paint;
+import android.net.Uri;
+import android.os.AsyncTask;
+import android.os.Environment;
 import android.os.Handler;
 import android.support.annotation.Nullable;
 import android.util.AttributeSet;
 import android.view.MotionEvent;
 import android.view.View;
+import android.widget.Toast;
 
 import com.yx.srtool.Activity.RocketDesignActivity;
 import com.yx.srtool.Activity.MainActivity;
+import com.yx.srtool.Adapter.GalaxyAdapter;
 import com.yx.srtool.Utils.FileUtil;
 import com.yx.srtool.Utils.Util;
 
 import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileOutputStream;
 import java.io.IOException;
+import java.text.SimpleDateFormat;
 import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
@@ -28,6 +37,7 @@
  */
 
 public class RocketView extends View{
+    private static Bitmap bitmap;
     private Paint Paint_Line//Paint_Line线
             ,Paint_Texe//Paint_Texe文字
             ,Paint_graph;//Paint_graph图形
@@ -74,7 +84,7 @@
             //Sprites_Bitmap : 分割精灵位图
             Sprites_Bitmap = getStringBitmapMap();
             //获取xml文件
-            String sprites_xml = FileUtil.read("PartList.xml");
+            String sprites_xml = FileUtil.read(MainActivity.path+"PartList.xml");
             //解析PartList.xml文件，XML
             PartList_xml = Util.parse_PartListXML(sprites_xml);
             //获取载具文件
@@ -119,11 +129,20 @@
     protected void onDraw(Canvas canvas) {
         initDraw(canvas);//初始化画布
         startDraw(canvas);
+
+
+        bitmap = Bitmap.createBitmap(canvas.getWidth(),canvas.getHeight(), Bitmap.Config.ARGB_8888);
+        Canvas mCanvas = new Canvas(bitmap);
+        //移动中心坐标
+        mCanvas.translate(getWidth()/2,getHeight()/2);
+        //缩放
+        mCanvas.scale(Multiple,Multiple);
+        startImage(mCanvas);
         super.onDraw(canvas);
-
     }
 
     private void initDraw(Canvas canvas) {
+
         //canvas.scale(1,-1);// 翻转Y轴
         //设置背景
         Bitmap Back = Bitmap.createScaledBitmap(Background, getWidth(), getHeight(), true);
@@ -349,6 +368,82 @@
     }
 
 
+    private void startImage(Canvas canvas) {
+        try {
+            for(int i =0;i<Ships_List.size();i++) {
+                Map<String,String> Ships_Parameter = Ships_List.get(i);
+                Map<String,String> Ships_png_path = PartList_xml.get(Ships_Parameter.get("partType"));
+                Bitmap Ships_Parameter_Bitmap,Ships_Parameter_Bitmap2 = null,Ships_Parameter_Bitmap3 = null;
+                boolean landerleg = false;
+                //判断当前是否是着陆架
+                if(Ships_png_path.get("sprite").toLowerCase().indexOf("landerlegpreview.png")!=-1){
+                    Ships_Parameter_Bitmap = Sprites_Bitmap.get("landerlegjoint.png");//头
+                    Ships_Parameter_Bitmap2 = Sprites_Bitmap.get("landerleglower.png");//支
+                    Ships_Parameter_Bitmap3 = Sprites_Bitmap.get("landerlegupper.png");//管
+                    landerleg = true;
+                }else {
+                    Ships_Parameter_Bitmap = Sprites_Bitmap.get(Ships_png_path.get("sprite").toLowerCase());//Bitmap,png格式
+                }
+
+                try {
+                    //计算角度
+                    float angle = (float) -(Float.parseFloat(Ships_Parameter.get("angle"))*((90/(Math.PI/2))));
+                    //图片中心坐标
+                    int offsetX = Ships_Parameter_Bitmap.getWidth() / 2;
+                    int offsetY = Ships_Parameter_Bitmap.getHeight() / 2;
+                    //图片变形
+                    Matrix matrix = new Matrix();
+                    Matrix matrix2 = new Matrix();
+                    //旋转： 往自身的移动半宽，半长
+                    matrix.postTranslate(-offsetX, -offsetY);
+                    //如果当前为着陆架
+                    if(landerleg){
+                        matrix2.postTranslate(- Ships_Parameter_Bitmap2.getWidth() / 2, - Ships_Parameter_Bitmap2.getWidth() / 2);
+                        matrix2.postRotate(180+angle);
+                    }
+                    //旋转
+                    matrix.postRotate(angle);
+                    //图片翻转
+                    if(Ships_Parameter.get("flippedX")!= null && Ships_Parameter.get("flippedY")!= null){
+                        if(Ships_Parameter.get("flippedX").equals("1")){
+                            matrix.postScale(-1, -1);
+                        }
+                        if(Ships_Parameter.get("flippedY").equals("1")){
+                            matrix.postScale(-1, -1);
+                        }
+                    }
+                    //判断当前是否未接入
+                    if(Ships_Parameter.get("Connections").equals("0")){
+                        //未接入组件
+                        //透明度
+                        //paint.setAlpha( 100 );
+                    }else {
+                        Paint paint = new Paint();
+                        //图片x,y
+                        float Bitmap_x = eventX0+2*(Float.parseFloat(Ships_Parameter.get("x"))*30);
+                        float Bitmap_y = eventY0-2*(Float.parseFloat(Ships_Parameter.get("y"))*30);
+                        //判断是否为着陆架
+                        if(landerleg){
+                            //设置Bitmap坐标
+                            matrix2.postTranslate(Bitmap_x, Bitmap_y);
+                            //渲染出图片
+                            canvas.drawBitmap(Ships_Parameter_Bitmap2, matrix2, paint);
+                            canvas.drawBitmap(Ships_Parameter_Bitmap3, matrix2, paint);
+                        }
+                        //设置Bitmap坐标
+                        matrix.postTranslate(Bitmap_x, Bitmap_y);
+                        //渲染出图片
+                        canvas.drawBitmap(Ships_Parameter_Bitmap, matrix, paint);
+                    }
+
+                } catch (Exception e) {
+                    e.printStackTrace();
+                }
+            }
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+    }
 
     /**
      * 屏幕事件
@@ -494,7 +589,7 @@
     private Map<String, Bitmap> getStringBitmapMap() throws Exception {
         Map<String,Bitmap> Sprites_Bitmap = new HashMap<>();
         //获取xml文件
-        String sprites = FileUtil.read("ShipSprites.xml");
+        String sprites = FileUtil.read(MainActivity.path+"ShipSprites.xml");
         //xml解析
         List<Map<String,String>> list = Util.parse_SpritesXML(sprites);
         //根据xml内的参数分割图片
@@ -520,4 +615,32 @@
         //Log.e("angle",angle+"");
         Ships_List.set(PartList_ID,Ships_Parameter);
     }
+
+    public static void saveimage(final RocketDesignActivity rocketDesignActivity) {
+        new AsyncTask(){
+            @Override
+            protected Object doInBackground(Object[] p1)
+            {
+
+                SimpleDateFormat sdf=new SimpleDateFormat("yyyyMMdd-HHmmss");
+                File file=new File(Environment.getExternalStorageDirectory(),sdf.format(System.currentTimeMillis())+".png");
+                try
+                {
+                    String stage="Saving";
+                    file.createNewFile();
+                    FileOutputStream fos=new FileOutputStream(file);
+                    if(bitmap.compress(Bitmap.CompressFormat.PNG,90,fos)){
+                        fos.flush();
+                        fos.close();
+                        stage="Saved As"+"/DCIM/"+sdf.format(System.currentTimeMillis())+".png";
+                        Uri uri = Uri.fromFile(file);
+                        rocketDesignActivity.sendBroadcast(new Intent(Intent.ACTION_MEDIA_SCANNER_SCAN_FILE, uri));
+                    }
+                }catch (IOException e){
+
+                }
+                return null;
+            }
+        }.execute();
+    }
 }
Index: app/src/main/java/com/yx/srtool/Utils/FileUtil.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- app/src/main/java/com/yx/srtool/Utils/FileUtil.java	(revision 47e437015c243ef1db46a1a21692136a088524ec)
+++ app/src/main/java/com/yx/srtool/Utils/FileUtil.java	(revision )
@@ -5,6 +5,7 @@
 import android.os.Environment;
 import android.os.Handler;
 import android.os.Message;
+import android.text.TextUtils;
 import android.util.Log;
 
 import com.yx.srtool.Activity.MainActivity;
@@ -37,10 +38,9 @@
     public static String read(String filepath) throws Exception {
         StringBuilder sb = new StringBuilder("");
         if(Environment.getExternalStorageState().equals(Environment.MEDIA_MOUNTED)){
-            String path = MainActivity.path+filepath;
             //filename = context.getExternalCacheDir().getAbsolutePath() + java.io.File.separator + path;
             //打开文件输入流
-            FileInputStream inputStream = new FileInputStream(path);
+            FileInputStream inputStream = new FileInputStream(filepath);
 
             Thread.sleep(200);
             byte[] buffer = new byte[1024];
@@ -204,6 +204,7 @@
     {
         try{
             InputStream fosfrom = new FileInputStream(fromFile);
+
             OutputStream fosto = new FileOutputStream(toFile);
             byte bt[] = new byte[1024];
             int c;
@@ -220,6 +221,20 @@
         }
     }
     /**
+     * 获取文件名及后缀
+     */
+    public static String getFileNameWithSuffix(String path) {
+        if(TextUtils.isEmpty(path)){
+            return "";
+        }
+        int start = path.lastIndexOf("/");
+        if (start != -1 ) {
+            return path.substring(start + 1);
+        } else {
+            return "";
+        }
+    }
+    /**
      * 删除方法 这里只会删除某个文件夹下的文件，如果传入的directory是个文件，将不做处理
      *
      * @param directory
